www.JamieOliver.com: updated selectors & tests

Mark-Howe opened the pull request:
- notes have a new selector (2x test cases)
- categories will no longer return duplicates
- Found a recipe returning a double up of ingredients & instructions, one set for Metric, another for Imperial. Default's to Metric if it happens.

Mark-Howe committed:
[www.JamieOliver.com] Removing a recipe that's no-longer online

Mark-Howe committed:
[www.JamieOliver.com] Don't allow duplicate Categories

Mark-Howe committed:
[www.JamieOliver.com] Notes have a new selector. Updated tests.

Mark-Howe committed:
[www.JamieOliver.com] Handle recipes listing Metric & Imperial ingredents/instructions
See https://www.jamieoliver.com/recipes/fish-recipes/jumbo-fish-fingers/ for an example, only one I can find.
This looks for the metric-specific ingredients & instructions, before returning the entire list

Mark-Howe commented on tests/data/results/www.jamieoliver.com/recipes-eggs-recipes-simple-cheese-omelette.php:
```diff
@@ -0,0 +1,47 @@
+        'When the omelette begins to cook and firm up, but still has a little raw egg on top, sprinkle over the cheese.',
+        'Using a spatula, ease around the edges of the omelette, then fold it over in half.',
+        'When it starts to turn golden brown underneath, remove the pan from the heat and slide the omelette onto a plate. Delicious with a tomato salad and wholemeal bread.',
+        'PRINT THIS RECIPE (UK) PRINT THIS RECIPE (USA)',
```
Maybe this should be caught and removed, it'd have to be via strpos or similar, they've chucked it into the method steps.

ssnepenthe replied:
Agreed - from a quick google search it looks like there are only a handful of recipes with a print link in the instructions - I would probably filter using stripos to avoid casing issues.

Mark-Howe replied:
Now filtering steps containing 'print this recipe'

ssnepenthe commented on tests/data/results/www.jamieoliver.com/recipes-eggs-recipes-simple-cheese-omelette.php:
```diff
@@ -0,0 +1,47 @@
+        'Family one-pan recipes',
+        'Mains',
+    ],
+    'cookingMethod' => '',
```
Missing values should be null instead of an empty string. Same goes for the fish fingers recipe

ssnepenthe commented:
Thanks for the PR! I added a couple of notes - can you check them and make a couple of quick fixes before I merge? Thanks again!

Mark-Howe committed:
[www.JamieOliver.com] Strip 'Print this...' from instructions
- Added a 2nd 'print this' test
- Replaced empty strings with null's in test results

Mark-Howe commented:
Should be good to go.

ssnepenthe commented:
Thanks! And sorry about the (massive) delay.
